File: src/main/java/com/meti/core/None.java
package com.meti.core;

import java.util.function.Function;

public class None<T> implements Option<T> {
    @Override
    public T unwrapOrPanic() {
        throw new RuntimeException("No unwrap present.");
    }

    @Override
    public boolean isPresent() {
        return false;
    }

    @Override
    public <R> Option<R> map(Function<T, R> mapper) {
        return new None<>();
    }

    @Override
    public T unwrapOrElse(T other) {
        return other;
    }

    @Override
    public Tuple<Boolean, T> toTuple(T other) {
        return new Tuple<>(false, other);
    }
}

File: src/main/windows/magma/option/None.c
struct None<T> implements Option<T>{};
/*Option_?*/ map_None<T> implements Option<T>(R mapper) {/*
		return new None<>();
	*/}
/*Option_?*/ flatMap_None<T> implements Option<T>(OptionR mapper) {/*
		return new None<>();
	*/}
T orElse_None<T> implements Option<T>(T other) {/*
		return other;
	*/}
